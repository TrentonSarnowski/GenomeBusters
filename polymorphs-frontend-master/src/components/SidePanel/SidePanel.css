.SidePanel {
  background-color: var(--lightest-color);
  height: 100%;
  margin: 0px;
  padding: 0px;
  position: absolute;
  text-align: center;
  top: 0px;
  width: 272px;
  z-index: 3;
}
/*box shadow*/

.SidePanel-cell{
  color: var(--lightest-color);
  background-color: var(--darkest-color);
  /*box-shadow: 2px 1px 5px rgba(0, 0, 0, 0.5);*/
}

.SidePanel-collapseBox {
  -webkit-transition: margin-left 0.2s;
  transition: margin-left 0.2s;
  width: 15px;
  height: 30px;
  margin-top: -50px;
  text-align: center;
  line-height: 26px;
  box-shadow: 2px 1px 5px rgba(0, 0, 0, 0.35);
}

.SidePanel-collapseBox.open {
  margin-left: 290px;
}

.SidePanel-collapseBox.closed {
  margin-left: 0px;
}

.SidePanel-mainBox {
  -webkit-transition: left 0.2s;
  transition: left 0.2s;
  text-align: left;
  margin: 0px; /*up right down left*/
  line-height: 44px;
  width: 272px;
  height: 100%;
  position: absolute;
  top: 0px;
  left: 0px;
  /*box-shadow: 0px 0px 3px 4px rgba(0, 2, 4, .5);*/
  /*box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.3);*/
  /*box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.2);*/
  
}

.SidePanel-mainBox.show {
	/*visibility: show;*/
	box-shadow: 2px 1px 5px rgba(0, 0, 0, 0.5);
}

.SidePanel-mainBox.hidden {
	left: -290px;
	box-shadow: 0 0 0 rgba(0, 0, 0, 0);
	/*visibility: hidden;*/
}

.SidePanel-hovBox{
  position: relative;
  background-color: var(--lightest-color);
  color: var(--darkest-color);
  text-align: center;
  height: 22px;
  line-height: 22px;
  width: 132px; /*put the size in units of REM?*/
  margin-top: -28px;
  border-style: solid;
  border-radius: 4px;
  box-shadow: 2px 1px 5px rgba(0, 0, 0, 0.2);
  font-size: .81rem;
  /*white with black border and shadow
	RATIO OF REM. 
  */
}

.SidePanel-hovBox::before{
  content:"";
  position: absolute;
  right: 100%;
  top: 26px;
  width: 0;
  height: 0;	
  border-top: 6px solid transparent;
  border-right: 10px solid var(--darkest-color);
  border-bottom: 6px solid transparent;
  margin-top: -21px;
  display: block;

}

.SidePanel-top {
  height: 20%;
  padding-left: 20px;
}

.SidePanel-mainBox-scroll {
  height: 540px;
  margin-top: 30px;
  overflow: auto;
  padding-left: 20px;
}

.SidePanel-mainBox-scroll ul {
  list-style-type: none;
  padding-left: 0px;
}


.SidePanel-mainBox-scroll ul li {
  padding-left: 5px;
}

.SidePanel-mainBox-scroll ul li:hover {
  background-color: #4E4E4E; 
}

.SidePanel-hovBox.yeshover{
	visibility: show;
}

.SidePanel-hovBox.nohover{
	visibility: hidden;
}

.SidePanel-hovBox.open{
  margin-left: 320px;
}

.SidePanel-hovBox.closed{
  margin-left: 30px;
}
