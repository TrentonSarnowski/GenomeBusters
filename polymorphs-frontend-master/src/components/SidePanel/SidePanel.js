import React, { Component } from 'react';
import './SidePanel.css';

class SidePanel extends Component {
  constructor(props) {
    super(props);
    this.state = {
    	menuActive: true,
    	hoverActive: false,
    }

    this.toggleMenu = this.toggleMenu.bind(this);
    this.activateHover = this.activateHover.bind(this);
    this.deactivateHover = this.deactivateHover.bind(this);
    this.gethovBoxPrompt = this.getHovBoxPrompt.bind(this);
    /*this.select = this.select.bind(this);*/
  }

  toggleMenu() {
    this.setState({menuActive: !this.state.menuActive});
  }
  activateHover() {
    this.setState({hoverActive: true});
  }
  deactivateHover(){
    this.setState({hoverActive: false});
  }
  getHovBoxPrompt(){
    if(this.state.menuActive){
      return <span>Collapse Side Panel</span>
    } else{
      return <span>Expand Side Panel</span>
    }
  }

  render() {
    return (
      <div className="SidePanel">
      	<div className={`SidePanel-mainBox ${this.state.menuActive ? 'show' : 'hidden'}`}>
        	<div className="SidePanel-mainBox SidePanel-cell">
            <div className="SidePanel-top">
              { this.props.filename? <h4>Viewing Genome: {this.props.filename}</h4> : <h4>Upload a file to start</h4> }
              { !this.props.gb_data && this.props.filename || this.props.isLoading ? <div>Loading...</div> : <div></div> }
              { this.props.gb_data && !this.props.isLoading? <div>Features found: {this.props.gb_data.sequence.features_found}</div> : <div></div> }
              { this.props.gb_data && !this.props.isLoading? <div>Sequence Length: {this.props.gb_data.sequence.sequence_length}</div> : <div></div> }
            </div>
            <div className="SidePanel-mainBox-scroll">
              <ul>
                {this.props.gb_data && this.props.gb_data.sequence ?
                  this.props.gb_data.sequence.features.map(function(gene) {
                    return gene.label ? <li key={gene.id}>{gene.label}</li> : "";
                  }, this)
                : ""}
              </ul>
            </div>
          </div>  
      	</div>
      	<div className={`SidePanel-collapseBox ${this.state.menuActive ? 'open' : 'closed'}`}>
     		<div className="SidePanel-collapseBox SidePanel-cell" onClick={this.toggleMenu} onMouseOver={this.activateHover} onMouseLeave={this.deactivateHover}>+</div>
     	</div>
     	<div className={`SidePanel-hovBox ${this.state.hoverActive ? 'yeshover' : 'nohover'} ${this.state.menuActive ? 'open' : 'closed'}`}>
     		{this.getHovBoxPrompt()}
     	</div>
    </div>
    );
  }

}

export default SidePanel;
