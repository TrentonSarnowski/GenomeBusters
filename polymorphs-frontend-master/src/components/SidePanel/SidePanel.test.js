import renderer from 'react-test-renderer';
import SidePanel from './SidePanel';
import React from 'react';
import ReactDOM from 'react-dom'
import ReactTestUtils from 'react-addons-test-utils'

test('Link renders correctly', () => {
  const tree = renderer.create(<SidePanel />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const renderer = ReactTestUtils.createRenderer();
  renderer.render(<SidePanel />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe('div');
});
