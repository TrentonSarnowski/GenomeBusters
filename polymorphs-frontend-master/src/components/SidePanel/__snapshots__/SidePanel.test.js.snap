exports[`test Link renders correctly 1`] = `
<div
  className="SidePanel">
  <div
    className="SidePanel-mainBox show">
    <div
      className="SidePanel-mainBox SidePanel-cell">
      bob
    </div>
  </div>
  <div
    className="SidePanel-collapseBox open">
    <div
      className="SidePanel-collapseBox SidePanel-cell"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}>
      +
    </div>
  </div>
  <div
    className="SidePanel-hovBox nohover open">
    <span>
      Collapse Side Panel
    </span>
  </div>
</div>
`;
